src/packages/emmett-postgresql/src/eventStore/projections/index.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  type Event,
+  type EventTypeOf,
+  type ReadEvent,
+} from '@event-driven-io/emmett';
+import pg from 'pg';
+import type { PostgresEventStoreOptions } from '../postgreSQLEventStore';
+
+export type PostgresProjectionHandler<EventType extends Event = Event> = (
+  connectionString: string,
+  client: pg.PoolClient,
+  events: ReadEvent<EventType>[],
+) => Promise<void> | void;
+
+export type ProjectionDefintion<EventType extends Event = Event> = {
+  type: 'inline';
+  name?: string;
+  canHandle: EventTypeOf<EventType>[];
```

Comment:
@oskardudycz just wondering, does it make sense to have a projection that can handle no events?

Shouldn't it be `NonEmptyArrayOf<EventTypeOf<EventType>>` :thinking: ?